tutorial/index.html

```diff
@@ -0,0 +1,412 @@
+    method to help you on the way.
+    </p>
+    <img src="tutorial/img/ready-set-go.jpg" alt="Go!">
+    <p>
+        Next: <a href="#tutorial/8" class="shortcut">Creating a New Flow</a>
+    </p>
+</div>
+<div>
+    <h2>Part 2: Creating a New Flow</h2>
+
+    <p>
+        Before we can expose a stream of filtered tweets on hash tag to the outside world, we'll first need something to
+        actually expose. This basically comes down to the fact that we need an extra flow, besides the flow for all tweets
+        and the flow for tweets of a user. Let's start by looking at the failing tests in <a
+            href="#code/src/test/scala/reactive/tweets/outgoing/TweetFlowSpec.scala" class="shortcut">TweetFlowSpec.scala</a>.
+    </p>
+<pre><code>"The flow for tweets with hash tag" should "only forward tweets with matching hash tag" in {
+  val hashTag = "shouldMatch"
+  val sut = tweetFlowOfHashTag(hashTag).runWith(TestSource.probe[Message], TestSink.probe[Message])
+  val (_, mockSink) = sut
+
+  val tweet = Tweet(User("test"), s"Hello World! #${hashTag}")
+  system.eventStream.publish(tweet)
+
+  mockSink.request(1)
+  mockSink.expectNext()
+  mockSink.expectNoMsg(noMessageTimeout)
+}</code></pre>
+    <p>
+        We mock out the <code>Source</code> and <code>Sink</code> to be able to control the input and output for a flow,
+        which makes testing a lot easier. We expect, given a flow that filters on hash tag X and putting a tweet on the
+        event bus with that same tag X, that we will actually receive that tweet in our mock sink. Because we only put
+        one tweet on the bus, we expect no more message after our first request.
+    </p>
+    <p>
+        There is another test to test the opposite scenario (putting a tweet on the bus with a non matching tag and
+        expecting not to see it arrive), but we'll not discuss since it is analogous to the test described above. Open
+        <a href="#code/src/main/scala/reactive/tweets/outgoing/TweetFlow.scala" class="shortcut">TweetFlow.scala</a> and
+        add an implementation on the spot marked with <code>???</code>. Of course your implementation should satisfy both
+        failing tests in <a
+            href="#code/src/test/scala/reactive/tweets/outgoing/TweetFlowSpec.scala" class="shortcut">TweetFlowSpec.scala</a>.
+    </p>
+    <p>
+        Next: <a href="#tutorial/9" class="shortcut">The Solution for this challenge (spoiler alert!)</a>
+    </p>
+</div>
+<div>
+    <h2>Part 2: Solution New Flow</h2>
+    <p>Drumroll...</p>
+    <code>def tweetFlowOfHashTag(hashTag: String) = tweetFlow(_.text contains hashTag)</code>
+    <p>
+        Your method should look something like this. Yes, it was that easy. Sorry.<br/><br/>
+        Note that the return type information can be omitted now. It was needed to make the tests compile. If you change
+        the implementation back to <code>???</code> and remove the return type information, you should see compile errors
```

Comment:
It's generally good practice to give all public methods an explicit return type. Is there a reason why we don't want it here? (Is it part of the Akka reactive streams DSL?)